src/main/java/sorald/cli/MineCommand.java

```diff
@@ -56,10 +55,35 @@
             split = ",")
     private List<Checks.CheckType> ruleTypes = new ArrayList<>();
 
+    @CommandLine.Option(
+            names = {Constants.ARG_HANDLED_RUES},
+            description =
+                    "When this argument is used, Sorald only mines violations of the rules that can be fixed by Sorald.")
+    private boolean handledRules;
+
     @Override
     public Integer call() throws Exception {
         List<? extends JavaFileScanner> checks = inferCheckInstances(ruleTypes);
 
+        if (handledRules) {
+            checks =
+                    checks.stream()
+                            .filter(
+                                    sc ->
+                                            Arrays.stream(
+                                                                    sc.getClass()
+                                                                            .getAnnotationsByType(
+                                                                                    Rule.class))
+                                                            .map(Rule::key)
+                                                            .map(Checks::stripDigits)
+                                                            .map(Integer::parseInt)
+                                                            .map(Processors::getProcessor)
+                                                            .filter(Objects::nonNull)
+                                                            .count()
+                                                    > 0)
+                            .collect(Collectors.toList());
+        }
```

Comment:
This can be simplified:

```suggestion
        if (handledRules) {
            checks =
                    checks.stream()
                            .filter(
                                    sc -> {
                                        int key =
                                                Integer.parseInt(Checks.getRuleKey(sc.getClass()));
                                        return Processors.getProcessor(key) != null;
                                    })
                            .collect(Collectors.toList());
        }
```

I would also move it into the `inferCheckInstances` method to not pollute the top level logic of the `call` method.

Reply:
Unrelated to this PR: It also seems to me like we should change the keys we use internally to match those in Sonar. It doesn't make sense for us to use integers instead of the strings that sonar-java uses, there's 0 benefit to that.

Reply:
> It doesn't make sense for us to use integers instead of the strings that sonar-java uses, there's 0 benefit to that

Great point!